docs/getting-started/quickstart-programming.md

```diff
@@ -79,178 +62,227 @@ exports.up = function(knex) {
-We'll do this with a [transactional function](../tutorials/transaction-tutorial.md).
-Copy this code into `src/operations.ts`:
+Now that we have `greetings` table, let's change our app to write to it. We'll do this with a [transactional function](../tutorials/transaction-tutorial.md). Change your `src/operations.ts` to contain:
 
 ```javascript
-import { TransactionContext, Transaction, HandlerContext, GetApi } from '@dbos-inc/dbos-sdk'
+import { TransactionContext, Transaction, HandlerContext, GetApi } from '@dbos-inc/dbos-sdk';
 import { Knex } from 'knex';
 
+interface GreetingRecord {
+  name: string;
+  note: string;
+}
+
 export class Greetings {
+  @GetApi('/') // Serve a quick readme for the app
+  static async readme(_ctxt: HandlerContext) {
+    const readme = '<html><body><p>' +
+      'Welcome! Visit the route /greeting/:name to be greeted!<br>' +
+      'For example, visit <a href="/greeting/dbos">/greeting/dbos</a>.<br>' +
+      '</p></body></html>';
+    return Promise.resolve(readme);
+  }
+
   @Transaction()
-  static async InsertGreeting(ctxt: TransactionContext<Knex>, friend: string, note: string) {
-    await ctxt.client.raw('INSERT INTO greetings (name, note) VALUES (?, ?)', [friend, note]);
-    ctxt.logger.info(`Greeting to ${friend} recorded in the database!`);
+  static async InsertGreeting(ctxt: TransactionContext<Knex>, gr: GreetingRecord) {
+    await ctxt.client('greetings').insert(gr);
+    ctxt.logger.info(`Greeting to ${gr.name} recorded in the database!`);
   }
 
   @GetApi('/greeting/:friend')
   static async Greeting(ctxt: HandlerContext, friend: string) {
     const noteContent = `Thank you for being awesome, ${friend}!`;
-    await ctxt.invoke(Greetings).InsertGreeting(friend, noteContent);
+    await ctxt.invoke(Greetings).InsertGreeting(
+      { name: friend, note: noteContent }
+    );
     return noteContent;
   }
 }
 ```
 
-The key elements of this code are:
-- We use the [`@Transaction`](../api-reference/decorators#transaction) decorator to define a [transactional function](../tutorials/transaction-tutorial.md) (`InsertGreeting`) that accesses the database from a managed client.
-- Inside `InsertGreeting`, we insert a row in the database with `ctxt.client.raw()`.
-- We add a line to `Greeting` invoking `InsertGreeting`.
+Here we define a `GreetingRecord` interface to match a row of data in our `greetings` table. We then define a `@Transaction` called `InsertGreeting` that puts a new `GreetingRecord` into `greetings`. Finally, we add a line to the GET API function `Greeting` to invoke `InsertGreeting` with the provided `name` and a welcoming `note`.
+
+Stop your app with CTRL+C. Rebuild with `npm run build` and start with `npx dbos start`. Make a few visits to the greeting URL in your browser, i.e. http://localhost:3000/greeting/Mike. With every new visit, the app should print this to the console:
+```
+[info]: Greeting to Mike recorded in the database! 
+```
+
+## Reading from the Database
```

Comment:
I'd remove this section. All it introduces is `readOnly`, which is a pure optimization that's too advanced for the programming guide in my opinion.

Reply:
Leaving in as discussed offline